### PR TITLE
Add ROS2 support

### PR DESCRIPTION
**Pull request by Timple**
Alternative to: https://github.com/hbanzhaf/steering_functions/pull/23

I tried to keep a minimal delta approach. All verification is still ROS1. 

ROS2 is capable of compiling pure cmake packages as long as there is a valid `package.xml`. Therefor most of the changes are there. The diff of the `package.xml` still looks a bit messy but the resulting file is quite readable imo.

Let me know what you think or if the version number needs to be any different.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,10 +19,8 @@ if(BUILD_WITH_ROS)
         geometry_msgs
         nav_msgs
         visualization_msgs
-        roslib
```
Comment (Timple) [inline CMakeLists.txt:17]: These two were unused even in ROS1

File: README.md [hunk 2]
```diff
@@ -55,50 +55,47 @@ The source code in this package is released under the Apache-2.0 License. For fu
 The [3rdparty-licenses.txt](3rd-party-licenses.txt) contains a list of other open source components included in this package.
 
 
-## Installation & Usage as a ROS package
-
 ### Dependencies
```
Comment (hbanzhaf) [inline README.md:58]: ## Dependencies

File: README.md [hunk 3]
```diff
@@ -55,50 +55,47 @@ The source code in this package is released under the Apache-2.0 License. For fu
...
 This package depends on the linear algebra library [Eigen], which can be installed by
 
     sudo apt-get install libeigen3-dev
 
-The [ROS] dependencies are listed in the package.xml and can be installed by
+The [ROS 1] or [ROS 2] dependencies are listed in the package.xml and can be installed by
 
     rosdep install steering_functions
```
Comment (hbanzhaf) [inline README.md:67]: ## Installation & Usage as a [ROS 1] or [ROS 2] package

File: README.md [hunk 4]
```diff
@@ -55,50 +55,47 @@ The source code in this package is released under the Apache-2.0 License. For fu
...
 
 
 ### Building
 
-To build this package from source, clone it into your catkin workspace and compile it in *Release* mode according to
+To build this package from source, clone it into your workspace and compile it in *Release* mode according to
 
-    cd catkin_ws/src
+    cd ws/src
```
Comment (hbanzhaf) [inline README.md:72]: How about `cd my_ws/src`?

File: README.md [hunk 5]
```diff
@@ -55,50 +55,47 @@ The source code in this package is released under the Apache-2.0 License. For fu
...
     git clone https://github.com/hbanzhaf/steering_functions.git
-    catkin build steering_functions -DCMAKE_BUILD_TYPE=Release
+    catkin build steering_functions -DCMAKE_BUILD_TYPE=Release  # [ROS 1]
```
Comment (hbanzhaf) [inline README.md:74]: Why do we need this line when the next line is for ROS 1 and ROS 2?

File: README.md [hunk 6]
```diff
@@ -55,50 +55,47 @@ The source code in this package is released under the Apache-2.0 License. For fu
...
+    colcon build --packages-up-to steering_functions --cmake-args -DCMAKE_BUILD_TYPE=Release  # [ROS 1] or [ROS 2]
 
-To launch a demo of the package, execute
+To launch a [ROS 1] demo of the package, execute
 
-    source catkin_ws/devel/setup.bash
+    source ws/devel/setup.bash
```
Comment (hbanzhaf) [inline README.md:79]: See above.

File: README.md [hunk 7]
```diff
@@ -55,50 +55,47 @@ The source code in this package is released under the Apache-2.0 License. For fu
...
+    <depend>steering_functions</depend>
+
+To link this library with another [ROS 1] package, add these lines to your package's CMakeLists.txt
+
+    find_package(catkin REQUIRED COMPONENTS steering_functions)
+    include_directories(${catkin_INCLUDE_DIRS})
+    target_link_libraries(${PROJECT_NAME}_node ${catkin_LIBRARIES})
 
-and the following lines to your package's package.xml
+Or [ROS 2]:
```
Comment (hbanzhaf) [inline README.md:95]: Or for [ROS 2]:

File: README.md [hunk 8]
```diff
@@ -107,13 +104,13 @@ Now the steering functions can be used in your package by including the appropri
...
 
-To build the unit tests, exectue
+To build the [ROS 1] unit tests, execute
 
     catkin build steering_functions -DCMAKE_BUILD_TYPE=Release --make-args tests
 
 To run a single test, e.g. the timing test, execute
 
-    cd catkin_ws/devel/lib/steering_functions
+    cd ws/devel/lib/steering_functions
```
Comment (hbanzhaf) [inline README.md:113]: See above.

File: package.xml [hunk 9]
```diff
@@ -1,36 +1,28 @@
...
-  <run_depend>roscpp</run_depend>
-  <run_depend>geometry_msgs</run_depend>
-  <run_depend>nav_msgs</run_depend>
-  <run_depend>visualization_msgs</run_depend>
-  <run_depend>roslib</run_depend>
-  <run_depend>costmap_2d</run_depend>
-  <run_depend>tf</run_depend>
-  <run_depend>eigen</run_depend>
+  <depend condition="$ROS_VERSION == 1">costmap_2d</depend>
+  <depend>eigen</depend>
```
Comment (hbanzhaf) [inline package.xml:18]: Can we move this one line up?

**Review by hbanzhaf — COMMENTED** [review]
I like it. Added only some minor comments.

**Comment by hbanzhaf** [comment]
@Rayman Does this PR also work for you?

File: package.xml (hunk 9 above)
Comment (Timple) [reply package.xml:18]: Sure thing, the idea was to be alphabetical

File: README.md (hunk 5 above)
Comment (Timple) [reply README.md:74]: Most ROS 1 user will use `catkin` for compilation. But colcon can be used for ROS 1 as well. So I think it's most clear to list them both and users can choose.

**Comment by Rayman** [comment]
> @Rayman Does this PR also work for you?

Yes, it's almost the same PR with the only difference that I ported the tests also to ROS 2.

File: README.md (hunk 5 above)
Comment (hbanzhaf) [reply README.md:75]: Okay, then let's just have '# ROS 1' (without the square brackets) and in the following line '# ROS 1 or ROS 2'

File: README.md [hunk 10]
```diff
@@ -55,50 +55,48 @@ The source code in this package is released under the Apache-2.0 License. For fu
...
 
-    source catkin_ws/devel/setup.bash
+    source my_ws/devel/setup.bash
     roslaunch steering_functions steering_functions.launch
 
 
 ### Linking
 
-To link this library with another [ROS] package, add these lines to your package's CMakeLists.txt
+Add this package as dependency to your package's `package.xml`
```
Comment (hbanzhaf) [inline README.md:86]: Just: "to your package.xml". We write package.xml in the section Dependencies also without apostrophes. I am flexible, we just have to be consistent.

**Review by hbanzhaf — COMMENTED** [review]
Looks good to me. Only some minor things left.

**Pull request merged**